runtime/include/tasks/qthreads/chpl-tasks-impl-fns.h

```diff
@@ -92,10 +92,10 @@ static inline chpl_qthread_tls_t* chpl_qthread_get_tasklocal(void)
                qthread_get_tasklocal(sizeof(chpl_qthread_tls_t));
         if (tls == NULL) {
             pthread_t me = pthread_self();
+            // if not process or comm thread, ok to return NULL
             if (pthread_equal(me, chpl_qthread_process_pthread)) {
                 tls = &chpl_qthread_process_tls;
             } else {
-                assert(chpl_qthread_comm_pthreads != NULL);
```

Comment:
I concur with @jabraham17's diagnosis: the form/placement of this assertion was rendered incorrect once #25317 made it possible to have `chpl_qthread_comm_num_pthreads == 0` in the steady-state.

I would have probably moved the assertion into the loop to preserve the spirit of the invariant being validated, or alternatively rewritten it to directly check the invariant, ie:
```
if (chpl_qthread_comm_num_pthreads > 0) assert(chpl_qthread_comm_pthreads != NULL);
```

The merged change no longer validates this invariant (which is notably established in a different file, runtime/src/tasks/qthreads/tasks-qthreads.c). Deleting the assertion entirely solves the immediate problem, but does not "prevent a null pointer dereference" (as claimed) if the invariant is someday violated.

Reply:
I agree with this sentiment and did consider when authoring this PR to move the assertion inside the loop, but ultimately decided not to because `chpl_qthread_comm_pthreads` should be non-null when `chpl_qthread_comm_num_pthreads>0`.

I think its reasonable to add the assertion back to the inside of the loop or to use your `if` statement. That more directly encodes the invariant into the code. I'll open a PR to do that (I slightly prefer the assert inside the loop approach), but that will not be merged until after the release

Reply:
See https://github.com/chapel-lang/chapel/pull/25335